test/test-context.js

```diff
@@ -6,22 +6,24 @@ test('Provides HTTP request headers with the context parameter', t => {
   t.plan(2);
   framework(context => {
     t.equal(typeof context.headers, 'object');
-  }, server => {
+  })
```

Comment:
Couldn't we use `await` instead?

Reply:
Either one works. Is there a reason you prefer `await`?

Reply:
I just like more linear, less indented code. But that's just my preference.

Reply:
Also doesn't it simplify exception handling? With `then()` you need also pass `rejected` cbk, right?

Reply:
> I just like more linear, less indented code. But that's just my preference.

It definitely looks better, and is probably easier to read and maintain in most cases. Unfortunately, `tape` doesn't work with `async`. If I change the test case to look something like this, the test will just hang.

```js
test('Exposes readiness URL', async t => {
  const server = framework(_ => '');
  request(server)
    .get('/health/readiness')
// etc
}
```
This is because `tape` doesn't know to wait for promise resolution in the `test()` callback. I think there are modules that help deal with this by wrapping all calls in a `promise.then.catch`, or I could roll my own. But I just figured it was easy enough to use `.then()`. :shrug:

Reply:
I see, then probably leave it as it is.